### DISCUSSION TITLE
Best standard Configuration?

### DISCUSSION THREAD

**Discussion by markmghali** [ESPresense/ESPresense#93] [Q&A]
I am wondering with all the new settings what is the best basic configuration for detecting android and iOS phones.  I have  4 esp32 running currently. I am seeing some ""jitter"" or hoping between rooms for my devices even though I am clearly next to a station.

Thank you for this amazing project

**Comment by DTTerastar** [comment]
mqtt_room is prone to jitter.  I see it as well.  It's programmed very simply and stupidly.  We need to send a ton of data very consistently and one little hiccup in data and mqtt_room puts you into a different room.  How is your uptime on the node that's having issues? This is what my status looks like currently:

![image](https://user-images.githubusercontent.com/1491145/137171665-d87d004f-cb70-4abe-8e17-87d462f69b3b.png)

Some of those office to kitchen are in fact me getting up and getting coffee.  Piper ipad is in a spot very in between 2 base station and flips constantly.

This is my config:
```
- platform: mqtt_room
  device_id: apple:watch6-1
  state_topic: espresense/rooms
  name: Darrell Watch
  timeout: 30
  away_timeout: 120

- platform: mqtt_room
  device_id: apple:watch2-7
  state_topic: espresense/rooms
  name: Heather Watch
  timeout: 30
  away_timeout: 120

- platform: mqtt_room
  device_id: roomAssistant:9d4cxx37d
  state_topic: espresense/rooms
  name: Darrell iPhone
  timeout: 30
  away_timeout: 120

- platform: mqtt_room
  device_id: roomAssistant:9004xx597
  state_topic: espresense/rooms
  name: Darrell Work Phone
  timeout: 30
  away_timeout: 120

- platform: mqtt_room
  device_id: roomAssistant:ad005xx3c00
  state_topic: espresense/rooms
  name: Darrells iPad Pro
  timeout: 30
  away_timeout: 120

- platform: mqtt_room
  device_id: apple:iPad4-2
  state_topic: espresense/rooms
  name: Piper iPad
  timeout: 30
  away_timeout: 120

- platform: mqtt_room
  device_id: tile:d15xxxce
  state_topic: espresense/rooms
  name: Darrell Wallet
  timeout: 120
  away_timeout: 300

- platform: mqtt_room
  device_id: tile:dc9xxx7d
  state_topic: espresense/rooms
  name: Darrell Keys
  timeout: 120
  away_timeout: 300

- platform: mqtt_room
  device_id: tile:ff9fxx36
  state_topic: espresense/rooms
  name: TV Remote
  timeout: 120
  away_timeout: 300

- platform: mqtt_room
  device_id: tile:c19xx09
  state_topic: espresense/rooms
  name: Darrell Air Pods
  timeout: 120
  away_timeout: 300

- platform: mqtt_room
  device_id: tile:dc2xxa
  state_topic: espresense/rooms
  name: Darrell Alt Keys
  timeout: 120
  away_timeout: 300

- platform: mqtt_room
  device_id: tile:f213xx0b
  state_topic: espresense/rooms
  name: Truck
  timeout: 120
  away_timeout: 300

- platform: mqtt_room
  device_id: tile:e3xxc7
  state_topic: espresense/rooms
  name: Heather Keys
  timeout: 120
  away_timeout: 300

- platform: mqtt_room
  device_id: apple:iphone12-8
  state_topic: espresense/rooms
  name: Heather iPhone
  timeout: 30
  away_timeout: 120
```

**Reply by markmghali** [reply]
I guess I was meaning more like the config on the espresense themselves all the new options and stuff? 

also my history doesn't look that clean much more jitter than that.

**Comment by JerseyJames** [comment]
So is there a solution to this @DTTerastar?
I would call my scenario 'bounce' really. A device (my iPhone in this case) will EXTREMELY temporarily appear in another room even though it hasn't moved. 
The screenshot shows just a 5 minute period. The time it spends in the 'other' room is actually immeasurable! ... ie. less than a second.
Eg. It may 'leave' the Office at 12:41:04 and then re-enter at 12:41:04.
I have tried specifying a value of a couple of seconds (2000ms) in 'Filtering - Skip reporting if message age is less that this (in milliseconds)' to no avail.
I may be able to fix this with automation config but I'm sure it's simple than that and I'm just missing a setting.

Any assistance gratefully received. Cool project.  :-) 

![image](https://user-images.githubusercontent.com/24659412/178260992-d7559603-99db-411a-884d-a82019afce56.png)

**Reply by DTTerastar** [reply]
> So is there a solution to this @DTTerastar? I would call my scenario 'bounce' really. A device (my iPhone in this case) will EXTREMELY temporarily appear in another room even though it hasn't moved. The screenshot shows just a 5 minute period. The time it spends in the 'other' room is actually immeasurable! ... ie. less than a second. Eg. It may 'leave' the Office at 12:41:04 and then re-enter at 12:41:04. I have tried specifying a value of a couple of seconds (2000ms) in 'Filtering - Skip reporting if message age is less that this (in milliseconds)' to no avail. I may be able to fix this with automation config but I'm sure it's simple than that and I'm just missing a setting.
> 
> Any assistance gratefully received. Cool project. :-)
> 
> ![image](https://user-images.githubusercontent.com/24659412/178260992-d7559603-99db-411a-884d-a82019afce56.png)

Some of those look like Home Assistant reboots.  MQTT room just picks the very first room that reports after a reboot.  The ultimate solution to this will be not using MQTT room to integrate, instead having the devices talk to each other and only switching once they agree on the new room.

**Reply by JerseyJames** [reply]
The ultimate solution sounds good and I can see from your description how Home Assistance reboots could cause the effect but (according to HA logs) that's not it in my case. Bear in mind the whole of the above shows just a 5 minute timespan with 18 jitters/bounces... two of which are just 5 seconds apart. Do you have any other ideas what could be causing it?

**Reply by erwinSmith757** [reply]
for me to overcome this, i use node red.... to explain it easily, i will simplify it.... i have inputs for each person location, and the BLE tracker update the input field for example tracker = kitchen.... then set input = kitchen.... but allow this only to happen, if there was motion in the room the last 30s... otherwise a persons location input cant be updated... and i also have other conditions, if a person is in bed, it also cant change...
